src/odemis/gui/comp/viewport.py

```diff
@@ -781,17 +783,108 @@ class AngularResolvedViewport(ViewPort):
 
     # Default class
     canvas_class = miccanvas.AngularResolvedCanvas
-    bottom_legend_class = None
+    bottom_legend_class = RadioLegend
+
+    def __init__(self, *args, **kwargs):
+        super(AngularResolvedViewport, self).__init__(*args, **kwargs)
+        # TODO why not sufficient to place here?
+        # bottom_legend_class = RadioLegend
 
     def setView(self, view, tab_data):
+        """
+        Set the microscope view that this viewport is displaying/representing
+        *Important*: Should be called only once, at initialisation.
+
+        :param view:(model.View)
+        :param tab_data: (model.MicroscopyGUIData)
+        """
         assert(self._view is None)
 
         self._view = view
         self._tab_data_model = tab_data
+        self.stream = None
 
         # canvas handles also directly some of the view properties
         self.canvas.setView(view, tab_data)
 
+        view.stream_tree.flat.subscribe(self.connect_stream)
+
+        self.bottom_legend.set_pol_callback(self.on_pol_change)
+
+    def connect_stream(self, _):
+        """ Find the most appropriate stream in the view to be displayed, and make sure the display
+        is updated when the stream is updated.
+
+        """
+
+        ss = self._view.getStreams()
+
+        if not ss:
+            stream = None
+        elif len(ss) > 1:
+            # => pick the first one playing
+            for s in ss:
```

Comment:
As for now you are no handling playing streams, you can drop this `for` loop, and only keep the `else`.